core-services/src/main/java/org/zalando/nakadi/service/publishing/EventPublisher.java

```diff
@@ -283,29 +284,36 @@ private void validate(final List<BatchItem> batch, final EventType eventType, fi
                                 "event.id", item.getEvent().getJSONObject("metadata").getString("eid"),
                                 "error", e.getMessage()));
                     }
-
                     throw e;
                 }
             }
+        } catch (final IOException ioe) {
+            throw new InternalNakadiException("Error closing active span scope", ioe);
         } finally {
             validationSpan.finish();
         }
     }
 
     private void submit(
-            final List<BatchItem> batch, final EventType eventType, final Span parentSpan, final boolean delete)
-            throws EventPublishingException {
+            final List<BatchItem> batch, final EventType eventType, final boolean delete)
+        throws EventPublishingException, InternalNakadiException {
         final Timeline activeTimeline = timelineService.getActiveTimeline(eventType);
         final String topic = activeTimeline.getTopic();
-        final Span publishSpan = TracingService.getNewSpanWithParent(parentSpan, "publishing_to_kafka")
-                .setTag(Tags.MESSAGE_BUS_DESTINATION.getKey(), topic);
-        try {
-            timelineService.getTopicRepository(eventType).syncPostBatch(topic, batch, eventType.getName(), delete);
+
+        final Span publishingSpan = TracingService.buildNewSpan("publishing_to_kafka")
+                .withTag(Tags.MESSAGE_BUS_DESTINATION.getKey(), topic)
+                .start();
+        try (Closeable ignored = TracingService.activateSpan(publishingSpan)) {
```

Comment:
Makes sense to utilize `TracingService.withActiveSpan()`, and log through `TracingService.log()` without providing a context